src/main/java/com/iexec/sms/secret/compute/TeeTaskComputeSecretService.java

```diff
@@ -116,33 +121,30 @@ public boolean encryptAndSaveSecret(OnChainObjectType onChainObjectType,
                                         String secretOwner,
                                         String secretKey,
                                         String secretValue) {
-        if (isSecretPresent(onChainObjectType, onChainObjectAddress, secretOwnerRole, secretOwner, secretKey)) {
+        try {
             final TeeTaskComputeSecret secret = TeeTaskComputeSecret
                     .builder()
                     .onChainObjectType(onChainObjectType)
                     .onChainObjectAddress(onChainObjectAddress)
                     .secretOwnerRole(secretOwnerRole)
                     .fixedSecretOwner(secretOwner)
                     .key(secretKey)
+                    .value(encryptionService.encrypt(secretValue))
                     .build();
-            log.info("Tee task compute secret already exists, can't update it." +
-                    " [secret:{}]", secret);
+            log.info("Adding new tee task compute secret [secret:{}]", secret);
+            int result = jdbcTemplate.update("INSERT INTO \"tee_task_compute_secret\" "
+                            + "(\"on_chain_object_type\", \"on_chain_object_address\", \"secret_owner_role\", \"fixed_secret_owner\", \"key\", \"value\") VALUES "
+                            + "(?, ?, ?, ?, ?, ?)",
+                    onChainObjectType.ordinal(), onChainObjectAddress.toLowerCase(), secretOwnerRole.ordinal(), secretOwner.toLowerCase(), secretKey, encryptionService.encrypt(secretValue));
+            if (result != 1) {
+                return false;
+            }
+            cacheSecretService.putSecretExistenceInCache(secret.getHeader(), true);
+            measuredSecretService.newlyAddedSecret();
+            return true;
+        } catch (DataAccessException e) {
+            log.error("{}", e.getMostSpecificCause().getMessage());
```

Comment:
Could we add a context element with the error ?

Reply:
I had the same thought, but after double-checking I found out that the error already states the context:
```
2024-02-08 10:34:30.072  INFO 35852 --- [    Test worker] c.i.s.s.c.TeeTaskComputeSecretService    : Adding new tee task compute secret [secret:com.iexec.sms.secret.compute.TeeTaskComputeSecret@8a3537d]
2024-02-08 10:34:30.161 ERROR 35852 --- [    Test worker] c.i.s.s.c.TeeTaskComputeSecretService    : Violation d'index unique ou clé primaire: "PUBLIC.PRIMARY_KEY_9 ON PUBLIC.tee_task_compute_secret(fixed_secret_owner, ""key"", on_chain_object_address, on_chain_object_type, secret_owner_role) VALUES ( /* key:1 */ '', '0', 'appaddress', 0, 0)"
Unique index or primary key violation: "PUBLIC.PRIMARY_KEY_9 ON PUBLIC.tee_task_compute_secret(fixed_secret_owner, ""key"", on_chain_object_address, on_chain_object_type, secret_owner_role) VALUES ( /* key:1 */ '', '0', 'appaddress', 0, 0)"; SQL statement:
INSERT INTO "tee_task_compute_secret" ("on_chain_object_type", "on_chain_object_address", "secret_owner_role", "fixed_secret_owner", "key", "value") VALUES (?, ?, ?, ?, ?, ?) [23505-222]
```